File: Year-2023/LeetCode-April-Challenge/OptimalPartitionofString.js
var partitionString = function (s) {
    let mp = {};
    let countUnique = 0;
    for(let x of s){
        if(!mp[x]){
            mp[x] ? mp[x]++: mp[x] = 1;
        } else{
            countUnique++;
            mp = {};
            mp[x] = 1;
        }
    }
    return ++countUnique;
};

let s = "abacaba";
console.log(partitionString(s));

File: Year-2023/LeetCode-April-Challenge/OptimalPartitionofString.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int partitionString(string s) {
        unordered_map<char,int> mp;
        int countUnique = 0;
        for(auto ch: s){
            if(!mp[ch]){
                mp[ch]++;
            }else{
                countUnique++;
                mp.clear();
                mp[ch]++;
            }
        }
        return ++countUnique;
    }
};
int main(){
    string s = "abacaba";
    Solution *S = new Solution();
    cout<<S->partitionString(s)<<endl;
    return 0;
}